odl/discr/discr_ops.py

```diff
@@ -130,110 +144,212 @@ def finite_diff(f, out=None, axis=0, dx=1.0, edge_order=2,
 
     method, method_in = str(method).lower(), method
     if method not in ('central', 'forward', 'backward'):
-        raise ValueError("method '{}' has to be central, forward or backward"
+        raise ValueError('method {} is not central, forward or backward'
                          ''.format(method_in))
 
+    if isinstance(padding, (int, float)) or padding in (None, 'edge'):
+        if isinstance(padding, int):
+            padding = float(padding)
+    else:
+        raise ValueError('padding value ({}) not valid'.format(padding))
+
+    if edge_order is None:
+        if method == 'central':
+            edge_order = 2
+        else:
+            edge_order = 1
+    else:
+        if edge_order not in [1, 2]:
+            raise ValueError('edge order ({0}) not valid'.format(edge_order))
+
+    if f_arr.shape[axis] == 2 and edge_order == 2:
```

Comment:
It's caught [here](https://github.com/odlgroup/odl/pull/230/files#diff-783324f48a9505cd5363ed78952719f9R123)